### DISCUSSION TITLE
server not starting

### DISCUSSION THREAD

**Discussion by Saphhmen** [pufferpanel/pufferpanel#1292] [Q&A]
I installed pufferpanel succesfuly, then i import the minecraft paper template, i create a server with it, pressed the install button and it installed sucessfuly. but when i tried to start it, it doesnt work

![image](https://github.com/pufferpanel/pufferpanel/assets/148562792/b46ba2ec-b8de-469a-b546-e9fc40627288)

**Comment by LucaBarbaLata** [comment]
do you have java installed?
what operating system are you using to host pufferpanel?

**Reply by Saphhmen** [reply]
i dont think i have java installed, but imusing ubuntu 22 (something, i forgot)
 i might wanna install java

**Reply by LucaBarbaLata** [reply]
sudo apt-get install openjdk-11-jre-headless -y

**Comment by LucaBarbaLata** [comment]
you need java in order for the server to run

**Comment by Saphhmen** [comment]
its still not working
![image](https://github.com/pufferpanel/pufferpanel/assets/148562792/a148c1c4-b688-4a25-8c88-1ee713129260)

**Comment by LucaBarbaLata** [comment]
sudo apt update
sudo apt install openjdk-17-jdk
then
java -version
and tell me the output

**Reply by Saphhmen** [reply]
i alr did install java 17

**Reply by Saphhmen** [reply]
![image](https://github.com/pufferpanel/pufferpanel/assets/148562792/d0696fd6-c257-4dad-8698-827e8c83e992)

**Comment by LucaBarbaLata** [comment]
try restarting ubuntu

**Comment by Saphhmen** [comment]
![image](https://github.com/pufferpanel/pufferpanel/assets/148562792/6772872c-6c69-46f5-9c04-6e1965afd99d)

**Comment by LucaBarbaLata** [comment]
you sure you are on ubuntu?

**Comment by LucaBarbaLata** [comment]
cuz i see some arch linux links there

**Comment by Saphhmen** [comment]
100%
should i install neofetch to prove

**Comment by LucaBarbaLata** [comment]
i trust you but idk why is there some links from arch linux. cuz pufferpanel dosent support arch as it says in their docs

**Reply by Saphhmen** [reply]
thats just the website
for some reason i named it "arclinux"

**Reply by LucaBarbaLata** [reply]
oh

**Reply by LucaBarbaLata** [reply]
so u use replit?

**Reply by Saphhmen** [reply]
ya

**Reply by Saphhmen** [reply]
![image](https://github.com/pufferpanel/pufferpanel/assets/148562792/78287fc1-db88-4abb-82b2-1fa3f13f45f3)

**Reply by LucaBarbaLata** [reply]
let me try it. brb

**Reply by LucaBarbaLata** [reply]
yeah i cant find a ubuntu image anyways

**Reply by Saphhmen** [reply]
i meaan..

**Reply by Saphhmen** [reply]
i could give u the code that managed me to run it on replit

**Reply by LucaBarbaLata** [reply]
please that could help me

**Reply by Saphhmen** [reply]
https://pasteio.com/xO3JVqLW7gq0

**Reply by Saphhmen** [reply]
after that choose "1", then run "apt install nano"
then run nano i.sh
then paste https://pasteio.com/xdd7D1Hk0cBz
save and exit, then run "bash i.sh"
then, it should setup the pufferpanel

**Reply by nepcore** [reply]
Why are you setting up a proot? If you need a proot environment because you don't have the necessary rights to install packages outside of that, then the service you're using clearly does not want you to do such things in which case what you're doing is at best dishonest and if you do have the needed rights to install external packages you don't need a proot environment  
The second script is even worse, for some reason it does a `chmod -R 777 /bin/systemctl` (besides other things) which should raise just about every possible red flag it could because that basically means literally anything and anyone managing to execute anything at all on your system can completely rewrite the main interface for interacting with systemd, the init system, which means that any such thing can take full control of the entire system, PufferPanel is also not that hard to install yourself, so why you need some third party script with zero sanity for that in the first place is also a mystery to me

**Reply by Saphhmen** [reply]
aight i got a new one, then i installed the official one
![image](https://github.com/pufferpanel/pufferpanel/assets/148562792/21921f30-b854-4265-a46c-85397ee1b75e)

**Comment by LucaBarbaLata** [comment]
try 
sudo apt install default-jdk
sudo apt install default-jre

**Comment by nepcore** [comment]
The entire "you need to install java" thing going on across all the responses here has been entirely incorrect for about a year now, since version 2.6.0 PufferPanel does handle java versions for you, no need to manually install anything, in fact manually installed java will more likely than not end up being missed, though even before PufferPanel dealing with java for you that response would have been incorrect, if the issue was that java wasn't found on the system it would tell you something like `exec: "java": executable file not found in $PATH`  
PufferPanel also doesn't care what exact distro you're using, if you managed to create a server it installed and runs just fine, be it on arch, ubuntu, fedora or whatever else you installed it on (supported versions regard the standard installation procedure, if you deal with it yourself or via a sensible AUR package we don't really care)  
What you actually need to do is the following:
1. check that the way you use replit is actually ok with their TOS, if you're using their remote IDE then that's not application hosting and should not be used as such, doing so violates their TOS, if you're using their actual application hosting services you might be fine, though I highly doubt you'll get _any_ amount of performance out of that, especially when looking at the pricing, it is very clearly not geared at game server hosting and I wouldn't be surprised if they do rule that out in their TOS
2. given that you got basically nothing on the web console, go ahead and check the PufferPanel logs (they're in `/var/log/pufferpanel`), those will contain more info on what exactly went wrong

**Reply by Saphhmen** [reply]
1. im 100% sure its okay with their TOS
2. i cant find the var folder nor pufferpanel

**Comment by Saphhmen** [comment]
boost

**Comment by LucaBarbaLata** [comment]
maybe there is somethng with the template that u used. maybe because replit dosent support ubuntu and it was forced to do so so it may have some limitations

**Reply by Saphhmen** [reply]
how about the problem
"System has not been booted with systemd as init system (PID 1). Can't operate. Failed to connect to bus: Host is down"

**Comment by Saphhmen** [comment]
bump
